### PR TITLE
Kaggarwal/add textgen cli

### PR DESCRIPTION
**Pull request by aggarwal-k**
# Description

# Checklist

- [ ] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [ ] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: cli/foundation-models/system/finetune/text-generation/download-dataset.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+args = parser.parse_args()
+
+# create the download directory if it does not exist
+if not os.path.exists(args.download_dir):
+    os.makedirs(args.download_dir)
+
+
+# import hugging face datasets library
+from datasets import load_dataset, get_dataset_split_names
+from functools import partial
```
Comment (jpmann) [inline cli/foundation-models/system/finetune/text-generation/download-dataset.py:28]: I don't see partial being used. Can we remove it if that is the case?

File: cli/foundation-models/system/finetune/text-generation/text-generation-pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+inputs:
+  compute_model_import: gpu-cluster-big
+  compute_preprocess: gpu-cluster-big
+  compute_finetune: gpu-cluster-big
+  compute_model_evaluation: gpu-cluster-big
+
+  # specify the foundation model available in the azureml system registry
+  mlflow_model_path: 
+    path: azureml://registries/azureml/models/Llama-2-7b/versions/5
```
Comment (jpmann) [inline cli/foundation-models/system/finetune/text-generation/text-generation-pipeline.yml:14]: Is it not possible to pass the latest version here similar to picking the latest pipeline component? azureml://registries/azureml/components/text_generation_pipeline/labels/latest

File: cli/foundation-models/system/finetune/text-generation/text-generation.sh [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+compute_cluster="gpu-cluster-big"
+# if above compute cluster does not exist, create it with the following vm size
+compute_sku="Standard_NC24rs_v3"
+# This is the number of GPUs in a single node of the selected 'vm_size' compute. 
+# Setting this to less than the number of GPUs will result in underutilized GPUs, taking longer to train.
+# Setting this to more than the number of GPUs will result in an error.
+# CLI command to fetch the number of GPUs
+# az ml compute list-sizes -g <resource_group_name> -w <workspace_name> --subscription <subscription_id> --query "[?name=='<compute_sku>'].{Name:name, Gpus:gpus}" --output table
+gpus_per_node=8
```
Comment (jpmann) [inline cli/foundation-models/system/finetune/text-generation/text-generation.sh:20]: `Standard_NC24rs_v3` has 4 gpus per node. Isn't it?

File: cli/foundation-models/system/finetune/text-generation/text-generation.sh [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+# This is the foundation model for finetuning
+model_name="Llama-2-7b"
+# using the latest version of the model - not working yet
```
Comment (jpmann) [inline cli/foundation-models/system/finetune/text-generation/text-generation.sh:23]: Is this comment still valid?

File: cli/foundation-models/system/finetune/text-generation/download-dataset.py (hunk 1 above)
Comment (aggarwal-k) [reply cli/foundation-models/system/finetune/text-generation/download-dataset.py]: Removed

File: cli/foundation-models/system/finetune/text-generation/text-generation.sh (hunk 3 above)
Comment (aggarwal-k) [reply cli/foundation-models/system/finetune/text-generation/text-generation.sh:20]: Changed to ND40 as that is the finetuning sku we are mentioning for text-gen

File: cli/foundation-models/system/finetune/text-generation/text-generation-pipeline.yml (hunk 2 above)
Comment (aggarwal-k) [reply cli/foundation-models/system/finetune/text-generation/text-generation-pipeline.yml:14]: no, tested it, its still not possible to give latest label

**Review by amangupta26 — APPROVED** [review]

**Pull request merged**